lib/Sugar.ml

```diff
@@ -291,49 +291,74 @@ let sequence (conf : Conf.t) cmts xexp =
 (* The sugar is different when used with the [functor] keyword. The syntax
    M(A : A)(B : B) cannot handle [_] as module name. *)
 let rec functor_type cmts ~for_functor_kw ({ast= mty; _} as xmty) =
+  let is_valid = function
+    | Some "_" -> false
+    | Some _ -> true
+    | None -> false
+  in
   let ctx = Mty mty in
   match mty with
-  | {pmty_desc= Pmty_functor (arg, arg_mty, body); pmty_loc; pmty_attributes}
-    when for_functor_kw
-         || (List.is_empty pmty_attributes && not (String.equal arg.txt "_"))
+  | { pmty_desc= Pmty_functor (Named (arg, arg_mty), body)
+    ; pmty_loc
+    ; pmty_attributes }
+    when for_functor_kw || (List.is_empty pmty_attributes && is_valid arg.txt)
     ->
-      let arg =
-        if String.equal "*" arg.txt then {arg with txt= ""} else arg
-      in
       Cmts.relocate cmts ~src:pmty_loc ~before:arg.loc ~after:body.pmty_loc ;
       let body = sub_mty ~ctx body in
       let xargs, xbody =
         match pmty_attributes with
         | [] -> functor_type cmts ~for_functor_kw body
         | _ -> ([], body)
       in
-      ((arg, Option.map arg_mty ~f:(sub_mty ~ctx)) :: xargs, xbody)
+      ((arg, Some (sub_mty ~ctx arg_mty)) :: xargs, xbody)
+  | {pmty_desc= Pmty_functor (Unit, body); pmty_attributes; _}
+    when for_functor_kw || List.is_empty pmty_attributes ->
+      let body = sub_mty ~ctx body in
+      let xargs, xbody =
+        match pmty_attributes with
+        | [] -> functor_type cmts ~for_functor_kw body
+        | _ -> ([], body)
+      in
+      (({txt= Some ""; loc= Location.none}, None) :: xargs, xbody)
   | _ -> ([], xmty)
 
 (* The sugar is different when used with the [functor] keyword. The syntax
    M(A : A)(B : B) cannot handle [_] as module name. *)
 let rec functor_ cmts ~for_functor_kw ~source_is_long ({ast= me; _} as xme) =
-  let valid_sugared_name arg =
-    not (String.equal arg.txt "_" && source_is_long me)
+  let is_valid = function
```

Comment:
They are all valid. I think the previous name (`valid_sugared_name`) was better.

Reply:
done